winch/codegen/src/codegen/env.rs

```diff
@@ -54,15 +80,13 @@ pub struct CalleeInfo {
 ///
 /// Contains all information about the module and runtime that is accessible to
 /// to a particular function during code generation.
-pub struct FuncEnv<'a, P: PtrSize> {
+pub struct FuncEnv<'a, 'b: 'a, 'c: 'b, P: PtrSize> {
```

Comment:
In my experience, when we start to get many lifetime parameters like this, it helps to give them more descriptive names.